plugins/modules/network/routeros/routeros_api.py

```diff
@@ -0,0 +1,483 @@
+
+    - name: Result prunt update status
+      ansible.builtin.debug:
+        msg: '{{ updateout }}'
+
+    - name: Remove ips -  stage 1 - query for '.id' "{{ ip2 }}" and "{{ ip3 }}"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        query: ".id address WHERE address == {{ item }}"
+      register: id_to_remove
+      loop:
+        - "{{ ip2 }}"
+        - "{{ ip3 }}"
+
+    # set fact for '.id' from 'query for {{ path }}'
+    - ansible.builtin.set_fact:
+        to_be_remove: "{{ to_be_remove |default([]) + [item['msg'][0]['.id']] }}"
+      loop: "{{ id_to_remove.results }}"
+
+    - name: Remove ips stage 1 - dump '.id'
+      ansible.builtin.debug:
+        msg: '{{ to_be_remove }}'
+
+    # Remove {{ 'rmips' }} with '.id' by 'to_be_remove' from query
+    - name: Remove ips -  stage 2 - remove "{{ ip2 }}" and "{{ ip3 }}" by '.id'
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        remove: "{{ item }}"
+      register: remove
+      loop: "{{ to_be_remove }}"
+
+    - name: Remove ips stage 2 dump result
+      ansible.builtin.debug:
+        msg: '{{ remove }}'
+
+    - name: Arbitrary command example "/system identity print"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "system identity"
+        cmd: "print"
+      register: cmdout
+
+    - name: Dump "/system identity print" output
+      ansible.builtin.debug:
+        msg: "{{ cmdout }}"
+'''
+
+RETURN = '''
+---
+message:
+    description: All outputs are in list with dictionary elements returned from RouterOS api.
+    type: list
+    returned: always
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import missing_required_lib
+from ansible.module_utils._text import to_native
+
+import ssl
+import traceback
+
+LIB_IMP_ERR = None
+try:
+    from librouteros import connect
+    from librouteros.query import Key
+    HAS_LIB = True
+except Exception as e:
+    HAS_LIB = False
+    LIB_IMP_ERR = traceback.format_exc()
+
+
+class ROS_api_module:
+    def __init__(self):
+        module_args = (dict(
+            username=dict(type='str', required=True),
+            password=dict(type='str', required=True, no_log=True),
+            hostname=dict(type='str', required=True),
+            port=dict(type='int'),
+            ssl=dict(type='bool', default=False),
+            path=dict(type='str', required=True),
+            add=dict(type='str'),
+            remove=dict(type='str'),
+            update=dict(type='str'),
+            cmd=dict(type='str'),
+            query=dict(type='str')))
+
+        self.module = AnsibleModule(argument_spec=module_args,
+                                    supports_check_mode=False)
```

Comment:
Since some of the options are mutually exclusive (add, remove, update, ...), you probably want to say that here. See for example https://github.com/ansible-collections/community.general/blob/34c3f9741d7f25b2d14882c0682529edb77f9c94/plugins/modules/packaging/os/pacman_key.py#L336 on how to do that. (You should also mention it in the documentation of these options.)

Reply:
> Since some of the options are mutually exclusive (add, remove, update, ...), you probably want to say that here. See for example https://github.com/ansible-collections/community.general/blob/34c3f9741d7f25b2d14882c0682529edb77f9c94/plugins/modules/packaging/os/pacman_key.py#L336 on how to do that. (You should also mention it in the documentation of these options.)

This is actually very good point ! 
Thaks

Reply:
fixed